lib/SILOptimizer/Utils/Existential.cpp

```diff
@@ -249,14 +249,25 @@ SILInstruction *swift::findInitExistential(Operand &openedUse,
 }
 
 /// Derive a concrete type of self and conformance from the init_existential
-/// instruction.
-/// If successful, initializes a valid ConformanceAndConcreteType.
-ConcreteExistentialInfo::ConcreteExistentialInfo(Operand &openedUse) {
+/// instruction.  If successful, initializes a valid ConformanceAndConcreteType.
+/// If hasSoleConformingType is set, then the concrete type determination is
+/// deferred to a later stage where cast instructionas have been added.
+ConcreteExistentialInfo::ConcreteExistentialInfo(
+    Operand &openedUse, const bool hasSoleConformingType)
+    : hasSoleConformingType(hasSoleConformingType) {
+
+  //  Bail, if this existential has a sole conforming type. If there is a sole
+  //  confirming type, we defer the determination of concrete type until cast
+  //  instructions are added in SILCombiner.
+  if (hasSoleConformingType)
```

Comment:
It's strange that you would intentionally construct an invalid `ConcreteExistentialInfo`, presumably to "fill it in" later. I'll have to review the related patches to understand why you need to do this.